### DISCUSSION TITLE
Erweiterung der seriellen Schnittstellen über i2c

### DISCUSSION THREAD

**Discussion by Autoschrauberix** [shining-man/bsc_fw#35] [Ideas]
wäre es denkbar die Anzahl der seriellen Schnittstellen zu erhöhen, z.B. mit einem SC16IS752 an i2c ?
Würde das hier [https://github.com/TD-er/SC16IS752](url) helfen. Ich bin leider noch ziemlicher Anfänger in dem Thema.
Gibt es eine Möglichkeit auf dem ersten BSC Board an den i2c Bus zu kommen ?

**Comment by shining-man** [comment]
Die I2C Schnittstelle ist auf allen Revisionen der Hardware auf Steckern. 
Bei der PCB v1.0 ist es auf dem X1 und bei der PCB v2.1 ist es auf dem J11 (6 pol. Stiftleiste vonre links auf der PCB).
Du wirst vmtl. die PCB v2.1 haben.

Der SC16IS752 passt von seiner maximalen Bussgeschwindigkeit nicht in das Konzept. Er kann maximal 400 kbit/s. 
Aktuell nutze ich den I2C Bus für das Display mit 1000 kbit/s.

Ich bin aber gerade dabei den BSC um eine Master/Slave funktionalität zu erweitern. Damit können dann mehre BSC kaskadiert werden.
Damit kann man dann auf 6 oder 9 serielle Schnittstellen erweitern, was ja genau das ist was du willst. Der Vorteil ist, dass auf dem BSC schon die galvanische Trennung usw. vorhanden ist.

Es gibt dazu schon ein Issue #32.

Wäre das für dich ausreichend? Wieviele weitere serielle Schnittstellen bräuchtest du?

**Reply by Autoschrauberix** [reply]
vielen Dank für die schnelle Antwort, das mit der Baudrate für das Display hatte ich übersehen. Die 400kbit/s hätten für die Aufgabe sicher gereicht.
Ich habe derzeit 5 Akkupacks mit JK BMS, max. sollen es mal 7 werden. Ich dachte einfach der Aufwand mit Kaskadierung ist höher, alle anderen Schnittstellen wären ja schon ausreichend vorhanden. Nochmal ein ESP braucht nur zusätzlich Energie usw. Dafür könnte man eine Art Redundanz einbauen, das hätte natürlich für eine Sicherheitseinrichtung schon etwas.
Ist es dann erforderlich dass ich zwei Boards aus der neuen Revision kaufe, wegen entspr. Schnittstelle ?
Ich schreib dann in Issue #32 weiter, da passt es besser.

**Reply by shining-man** [reply]
Bei der Kaskadierung ist an den Slaves das Wlan und Bluetooth inaktiv, was die meiste Energie braucht.
Der Aufwand die Software anzupassen ist bei beiden Varianten ziemlich ähnlich. Beim SC16IS752 bräuchtest du erst noch Hardware, außer du willst einen fliegenden Aufbau. Hiervon bin ich aber aboslut kein Fan. Daher ist es meines erachtens in Summe der wenigste Aufwand das ganze zu kaskadieren.
Wenn du 4 weitere Schnittstellen brauchst, dann brauchst du zwei zusätzliche BSC.
Von den Kosten her wird es auch nicht viel unterschied machen, wenn du ein PCB mit galvanischer Trennung und dem SC16IS752 machst.
